ambry-router/src/main/java/com.github.ambry.router/GetBlobInfoOperation.java

```diff
@@ -370,5 +441,35 @@ private void checkAndMaybeComplete() {
       NonBlockingRouter.completeOperation(null, getOperationCallback, operationResult, e);
     }
   }
+
+  /**
+   * @return true if the content is being decrypted
+   */
+  private boolean onDecryptMode() {
```

Comment:
Yep, maybe we could just do a `getState() == DECRYPTING`

Reply:
I thought about this and I feel that we do not need a `Decrypting` state at all, `InProgress` should cover that.

I understand that `Decrypting` gives more information, but within the code I think it makes things more complex (yet another state to check before doing N number of things), and it does not add a lot of value, so I would prefer keeping things simple.

The way I have come to look at the states is that these are really to benefit the `RequestResponseHandler` thread (and for Puts, also the `ChunkFillerThread`). And as far as the `RequestResponseHandler` is concerned, decryption is an internal thing for the chunk. It is part of the "progress" it is making. The states therefore mean the following:

```
Free, -> can be assigned to fetch a new chunk.
Ready, -> has been assigned responsibility of a new chunk and can be polled to send out requests.
InProgress -> is awaiting responses (and/or decryption), can be polled to send more requests or handle expirations
Complete -> responses have arrived (and/or decrypted where necessary) and the operation on the chunk has succeeded or failed.
```

Please see if there is any other value in this state, or whether it can simply be tracked internally.

Reply:
Based on our offline discussions, I have removed the addition of new states from both GetBlobInfoOperation and GetBlobOperation